src/main/java/uk/gov/companieshouse/limitedpartnershipsapi/model/dto/validator/NationalityValidator.java

```diff
@@ -0,0 +1,260 @@
+        modifiableListOfNationalities.add("Solomon Islander");
+        modifiableListOfNationalities.add("Somali");
+        modifiableListOfNationalities.add("South African");
+        modifiableListOfNationalities.add("South Korean");
+        modifiableListOfNationalities.add("South Sudanese");
+        modifiableListOfNationalities.add("Spanish");
+        modifiableListOfNationalities.add("Sri Lankan");
+        modifiableListOfNationalities.add("St Helenian");
+        modifiableListOfNationalities.add("St Lucian");
+        modifiableListOfNationalities.add("Stateless");
+        modifiableListOfNationalities.add("Sudanese");
+        modifiableListOfNationalities.add("Surinamese");
+        modifiableListOfNationalities.add("Swazi");
+        modifiableListOfNationalities.add("Swedish");
+        modifiableListOfNationalities.add("Swiss");
+        modifiableListOfNationalities.add("Syrian");
+        modifiableListOfNationalities.add("Taiwanese");
+        modifiableListOfNationalities.add("Tajik");
+        modifiableListOfNationalities.add("Tanzanian");
+        modifiableListOfNationalities.add("Thai");
+        modifiableListOfNationalities.add("Togolese");
+        modifiableListOfNationalities.add("Tongan");
+        modifiableListOfNationalities.add("Trinidadian");
+        modifiableListOfNationalities.add("Tristanian");
+        modifiableListOfNationalities.add("Tunisian");
+        modifiableListOfNationalities.add("Turkish");
+        modifiableListOfNationalities.add("Turkmen");
+        modifiableListOfNationalities.add("Turks and Caicos Islander");
+        modifiableListOfNationalities.add("Tuvaluan");
+        modifiableListOfNationalities.add("Ugandan");
+        modifiableListOfNationalities.add("Ukrainian");
+        modifiableListOfNationalities.add("Uruguayan");
+        modifiableListOfNationalities.add("Uzbek");
+        modifiableListOfNationalities.add("Vatican citizen");
+        modifiableListOfNationalities.add("Venezuelan");
+        modifiableListOfNationalities.add("Vietnamese");
+        modifiableListOfNationalities.add("Vincentian");
+        modifiableListOfNationalities.add("Wallisian");
+        modifiableListOfNationalities.add("Welsh");
+        modifiableListOfNationalities.add("Yemeni");
+        modifiableListOfNationalities.add("Zambian");
+        modifiableListOfNationalities.add("Zimbabwean");
+        nationalities = Collections.unmodifiableList(modifiableListOfNationalities);
+    }
+
+    @Override
+    public boolean isValid(Object object, ConstraintValidatorContext constraintValidatorContext) {
+        if (object instanceof GeneralPartnerDataDto) {
+            GeneralPartnerDataDto dto = (GeneralPartnerDataDto)object;
+            String nationality1 = dto.getNationality1();
+            String nationality2 = dto.getNationality2();
+            if (StringUtils.isNotBlank(nationality2)) {
+                return !nationality1.equals(dto.getNationality2()) &&
```

Comment:
on line 252 you can use the nationality2 variable instead of dto.getNationality2()

Reply:
Updated